source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -7,9 +7,97 @@ namespace Extensions {
 namespace Clusters {
 namespace DynamicForwardProxy {
 
-// TODO(mattklein123): Make sure that the cluster's hosts display their host name in admin output.
-// TODO(mattklein123): Allow customizing TLS on a per-host basis. For example, setting SNI and
-//                     doing certificate validation.
+class ClusterInfoWithOverridenTls : public Upstream::ClusterInfo {
+public:
+  ClusterInfoWithOverridenTls(const Upstream::ClusterInfoConstSharedPtr& real_cluster_info,
+                              Network::TransportSocketFactoryPtr&& transport_socket_factory)
+      : real_cluster_info_(real_cluster_info),
+        transport_socket_factory_(std::move(transport_socket_factory)) {}
+
+  // Upstream::ClusterInfo
+  bool addedViaApi() const override { return real_cluster_info_->addedViaApi(); }
```

Comment:
Optional, do you think it's worth having a base wrapper class, and then having ClusterInfoWithOverridenTls override that and the few functions it needs to?  Pros it's reusable and easier to see where ClusterInfoWithOverridenTls is doing real work.  Cons, extra code which may not get reused.

Reply:
Sure, I can do that. I will move the base wrapper class in with the other logical host stuff.